actors/evm.go

```diff
@@ -51,22 +51,25 @@ func (p *ActorParser) resurrect(raw []byte) (map[string]interface{}, error) {
 func (p *ActorParser) invokeContract(rawParams, rawReturn []byte, msgCid cid.Cid, ethLogs []types.EthLog) (map[string]interface{}, error) {
 	metadata := make(map[string]interface{})
 	reader := bytes.NewReader(rawParams)
-	var params abi.CborBytes
+	metadata[parser.ParamsKey] = parser.EthPrefix + hex.EncodeToString(rawParams)
+	metadata[parser.ReturnKey] = parser.EthPrefix + hex.EncodeToString(rawReturn)
 
+	var params abi.CborBytes
 	if err := params.UnmarshalCBOR(reader); err != nil {
 		p.logger.Sugar().Warn(fmt.Sprintf("error deserializing rawParams: %s - hex data: %s", err.Error(), hex.EncodeToString(rawParams)))
-		metadata[parser.ParamsKey] = parser.EthPrefix + hex.EncodeToString(rawParams)
-	} else {
+	}
+
+	if reader.Len() == 0 {
```

Comment:
This indicates that the reader was fully read, right?

Reply:
Can we add a comment here for future developers that review this? even for us, in the future!